package-res/resources/web/pentaho/visual/base/View.js

```diff
@@ -22,89 +22,180 @@ define([
   "use strict";
 
   /**
-   * @name View
+   * @name pentaho.visual.base.View
    * @memberOf pentaho.visual.base
    * @class
    * @extends pentaho.lang.Base
    * @abstract
    * @classDesc This is the base class for visualizations.
    *
+   * A container is expected to instantiate a `View` with a reference to a `Model` instance,
+   * which may not be immediately valid.
+   *
+   * Over time, the container mutates the `Model` instance and triggers events.
+   * In response, the method [_onChange]{@link pentaho.visual.base.View#_onChange} is
+   * invoked to process the events, which may cause the `View` to update itself.
+   *
    * @description Creates a base `View` instance.
+   *
    * @constructor
-   * @throws {Error} A valid DOM element must be passed to the constructor.
    * @param {HTMLElement} element The DOM element where the visualization should render.
+   * An error is thrown if this is not a valid DOM element.
    * @param {pentaho.visual.base.Model} model The base visualization `Model`.
+   * @throws {Error} Argument required. The constructor must be invoked with two arguments.
+   * @throws {Error} Argument invalid. A valid DOM element must be passed to the constructor.
    */
 
   var View = Base.extend(/** @lends pentaho.visual.base.View# */{
     constructor: function(element, model) {
-      if(!isElement(element)) {
+      if(arguments.length === 0)
```

Comment:
This is original. However, I don't think this is a good idea. We should have a single error message for a single error condition and in this code we're, in practice, validating the requiredness of `model` argument twice. Also, we should try to make the cases `nully` and absent/missing to mean and _result_ the same, whenever possible.